.github/workflows/ci.yml

```diff
@@ -98,10 +98,40 @@ jobs:
       - uses: Swatinem/rust-cache@v2.0.0
       - name: Install nextest
         uses: taiki-e/install-action@nextest
-      - name: Compile tests before running
-        run: cargo test --workspace --no-run
-      - name: Test execution
-        run: cargo nextest run --profile ci
+      - name: Build and archive tests
+        run: cargo nextest archive --archive-file nextest-archive.tar.zst
+      - name: Upload archive to workflow
+        uses: actions/upload-artifact@v3
+        with:
+          name: nextest-archive-${{ matrix.os }}
+          path: nextest-archive.tar.zst
+
+  run-tests-partitioned:
+    name: Run tests with partitioning
+    strategy:
+      matrix:
+        partition: [1, 2]
+        os: [ubuntu-latest, macos-12]
+    # XXX: Unfortunately github actions do not support matrix in 'needs' section
```

Comment:
🙃 I don't fully understand what to do with this comment. Can we just remove it or make it clearer?

Reply:
removed the comment; it turned out not essential for writing a working GH action.